spec/javascripts/stimulus/remote_toggle_controller_test.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { Application } from "stimulus";
+import remote_toggle_controller from "../../../app/webpacker/controllers/remote_toggle_controller";
+
+describe("RemoteToggleController", () => {
+  describe("#toggle", () => {
+    beforeEach(() => {
+      document.body.innerHTML = `
+        <div data-controller="remote-toggle" data-remote-toggle-selector-value="#content">
+          <button id="remote-toggle" data-action="click->remote-toggle#toggle"></button>
+          <button id="remote-toggle-with-chevron" data-action="click->remote-toggle#toggle">
+            <i class="icon-chevron-down" data-remote-toggle-target="chevron"></i>
+          </button>
+        </div>
+        <div id="content">...</div>
+      `;
+
+      const application = Application.start();
+      application.register("remote-toggle", remote_toggle_controller);
+    });
+
+    it("clicking a toggle switches the visibility of the :data-remote-toggle-selector element", () => {
+      const button = document.getElementById("remote-toggle");
+      const content = document.getElementById("content");
+      expect(content.style.display).toBe("");
+
+      button.click();
+
+      expect(content.style.display).toBe("none");
+
+      button.click();
+
+      expect(content.style.display).toBe("block");
+    });
+
+    /* skipping, this test passes when it's the only test in this file but not otherwise? */
+    it.skip("clicking a toggle with a chevron icon switches the visibility of content and the direction of the icon", () => {
```

Comment:
I think I figured it out. The `beforeEach` block gets called for each test but the window/browser object is not reset between each one, which means we actually initialize the whole StimulusJS framework once in the first test and then a second time in the second test. You can also just change the order of these two tests and whichever is first will pass, and whichever is second will fail. 

Shifting the initialization part into a `beforeAll` block fixes it. Like this: 

```js
beforeAll(() => {
  const application = Application.start();
  application.register("remote-toggle", remote_toggle_controller);
});
beforeEach(() => {
  document.body.innerHTML = `
      <div data-controller="remote-toggle" data-remote-toggle-selector-value="#content">
        <button id="remote-toggle" data-action="click->remote-toggle#toggle"></button>
        <button id="remote-toggle-with-chevron" data-action="click->remote-toggle#toggle">
          <i class="icon-chevron-down" data-remote-toggle-target="chevron"></i>
        </button>
      </div>
      <div id="content">...</div>
    `;
});
```